Index: chromium-122.0.6261.111/build/config/rust.gni
===================================================================
--- chromium-122.0.6261.111.orig/build/config/rust.gni
+++ chromium-122.0.6261.111/build/config/rust.gni
@@ -214,6 +214,8 @@ if (is_linux || is_chromeos) {
     } else {
       rust_abi_target = "arm-unknown-linux-gnueabi" + float_suffix
     }
+  } else if (current_cpu == "riscv64") {
+    rust_abi_target = "riscv64gc-unknown-linux-gnu"
   } else {
     # Best guess for other future platforms.
     rust_abi_target = current_cpu + "-unknown-linux-gnu"
