diff --git a/src/go/conf/zabbix_agent2.conf b/src/go/conf/zabbix_agent2.conf
index 6fdd050..fe48861 100644
--- a/src/go/conf/zabbix_agent2.conf
+++ b/src/go/conf/zabbix_agent2.conf
@@ -10,6 +10,8 @@
 # Default:
 # PidFile=/tmp/zabbix_agent2.pid
 
+PidFile=/run/zabbix/zabbix_agent2.pid
+
 ### Option: LogType
 #	Specifies where log messages are written to:
 #		system  - syslog
@@ -27,6 +29,8 @@
 # Default:
 # LogFile=/tmp/zabbix_agent2.log
 
+LogFile=/var/log/zabbix/zabbix_agent2.log
+
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
 #	0 - disable automatic log rotation.
@@ -139,7 +143,7 @@ ServerActive=127.0.0.1
 # Default:
 # Hostname=
 
-Hostname=Zabbix server
+# Hostname=Zabbix server
 
 ### Option: HostnameItem
 #	Item used for generating Hostname if it is undefined. Ignored if Hostname is defined.
@@ -149,6 +153,8 @@ Hostname=Zabbix server
 # Default:
 # HostnameItem=system.hostname
 
+HostnameItem=system.run[hostname -s]
+
 ### Option: HostMetadata
 #	Optional parameter that defines host metadata.
 #	Host metadata is used at host auto-registration process.
@@ -293,6 +299,8 @@ Hostname=Zabbix server
 # Default:/tmp/agent.plugin.sock
 # PluginSocket=
 
+PluginSocket=/run/zabbix/agent.plugin.sock
+
 ####### USER-DEFINED MONITORED PARAMETERS #######
 
 ### Option: UnsafeUserParameters
@@ -334,7 +342,7 @@ Hostname=Zabbix server
 # Default:
 # ControlSocket=
 
-ControlSocket=/tmp/agent.sock
+ControlSocket=/run/zabbix/zabbix_agent2.sock
 
 ####### TLS-RELATED PARAMETERS #######
 
