--- a/conf/zabbix_agentd.conf	2026-05-10 11:51:52.083459623 +0300
+++ b/conf/zabbix_agentd.conf	2026-05-10 11:55:40.916815948 +0300
@@ -10,6 +10,8 @@
 # Default:
 # PidFile=/tmp/zabbix_agentd.pid
 
+PidFile=/run/zabbix/zabbix_agentd.pid
+
 ### Option: LogType
 #	Specifies where log messages are written to:
 #		system  - syslog
@@ -27,7 +29,7 @@
 # Default:
 # LogFile=
 
-LogFile=/tmp/zabbix_agentd.log
+LogFile=/var/log/zabbix/zabbix_agentd.log
 
 ### Option: LogFileSize
 #	Maximum size of log file in MB.
@@ -177,7 +179,7 @@ ServerActive=127.0.0.1
 # Default:
 # Hostname=
 
-Hostname=Zabbix server
+# Hostname=Zabbix server
 
 ### Option: HostnameItem
 #	Item used for generating Hostname if it is undefined. Ignored if Hostname is defined.
@@ -187,6 +189,8 @@ Hostname=Zabbix server
 # Default:
 # HostnameItem=system.hostname
 
+HostnameItem=system.hostname
+
 ### Option: HostMetadata
 #	Optional parameter that defines host metadata.
 #	Host metadata is used at host auto-registration process.
