--- a/pkgmk.in	2023-12-02 00:44:50.922529207 +0200
+++ b/pkgmk.in	2024-06-11 20:37:13.229228265 +0300
@@ -324,7 +324,11 @@ check_signature() {
 		srcind=$(( srcind+1 ))
 		done
 
-		for FILE in "$PKGMK_PKGFILE" "$PKGMK_FOOTPRINT" "${source[@]}"; do
+		if [ ! -f $PKGMK_ROOT/$PKGMK_FOOTPRINT -a ! -z $MULTI_ARCH ]; then
+		    PKGMK_FOOTPRINT=("${PKGMK_FOOTPRINT}.riscv64" "${PKGMK_FOOTPRINT}.aarch64")
+		fi
+
+		for FILE in "$PKGMK_PKGFILE" "${PKGMK_FOOTPRINT[@]}" "${source[@]}"; do
 			ln -s $(get_filename -a $FILE) .
 		done
 
@@ -397,7 +401,11 @@ make_signature() {
 		srcind=$(( srcind+1 ))
 	done
 
-	for FILE in "$PKGMK_PKGFILE" "$PKGMK_FOOTPRINT" "${source[@]}"; do
+	if [ ! -f $PKGMK_ROOT/$PKGMK_FOOTPRINT -a ! -z $MULTI_ARCH ]; then
+	    PKGMK_FOOTPRINT=("${PKGMK_FOOTPRINT}.riscv64" "${PKGMK_FOOTPRINT}.aarch64")
+	fi
+
+	for FILE in "$PKGMK_PKGFILE" "${PKGMK_FOOTPRINT[@]}" "${source[@]}"; do
 		LOCAL_FILENAMES="$LOCAL_FILENAMES `get_filename $FILE`"
 	done
 
@@ -419,7 +427,11 @@ make_signature() {
 		error "Creating signature failed"
 		exit $E_SIGNATURE
 	else
-		mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}
+		if [ -f ${PKGMK_SIGNATURE}.${MULTI_ARCH} ]; then
+		    mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}.${MULTI_ARCH}
+		else
+		    mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}
+		fi
 	fi
 }
 
@@ -442,7 +454,11 @@ refresh_signature() {
 		error "Refreshing signature failed"
 		exit $E_SIGNATURE
 	else
-		mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}
+		if [ -f ${PKGMK_SIGNATURE}.${MULTI_ARCH} ]; then
+		    mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}.${MULTI_ARCH}
+		else
+		    mv ${PKGMK_SIGNATURE}.tmp ${PKGMK_SIGNATURE}
+		fi
 	fi
 }
 
@@ -502,6 +518,10 @@ check_footprint() {
 
 	cd $PKGMK_ROOT
 
+	if [ ! -f $PKGMK_FOOTPRINT -a ! -z $MULTI_ARCH ]; then
+	    PKGMK_FOOTPRINT+=".$MULTI_ARCH"
+	fi
+
 	if [ -f $TARGET ]; then
 		make_footprint > $FILE.footprint
 		if [ -f $PKGMK_FOOTPRINT ]; then
@@ -688,6 +708,10 @@ update_footprint() {
 		exit 1
 	fi
 
+	if [ ! -f $PKGMK_ROOT/$PKGMK_FOOTPRINT -a ! -z $MULTI_ARCH ]; then
+	    PKGMK_FOOTPRINT+=".${MULTI_ARCH}"
+	fi
+
 	check_file "$PKGMK_FOOTPRINT"
 	make_footprint > $PKGMK_FOOTPRINT
 	touch $TARGET
@@ -961,6 +985,8 @@ trap "interrupted" SIGHUP SIGINT SIGQUIT
 
 export LC_ALL=C.UTF-8
 
+MULTI_ARCH=${MULTI_ARCH:-$( [ $( uname -m ) == "aarch64" -o $( uname -m ) == "riscv64" ] && echo "$( uname -m)" || echo "" )}
+
 readonly PKGMK_VERSION="#VERSION#"
 readonly PKGMK_COMMAND="$0"
 readonly PKGMK_ROOT="$PWD"
