--- a/rc	2025-01-16 20:19:46.051064519 +0200
+++ b/rc	2025-01-16 20:21:36.711075637 +0200
@@ -118,7 +118,7 @@ printinfo "\nDeleting temporary files.."
 
 # Set kernel variables
 printinfo "\nSetting kernel variables.."
-/sbin/sysctl -p > /dev/null || printf "$(BOLD "$(RED)")" "[ERROR]"
+/sbin/sysctl -e --system > /dev/null || printf "$(BOLD "$(RED)")" "[ERROR]"
 
 # Update shared library links
 printinfo "\nUpdating shared library links.."
