--- a/rc	2025-05-07 15:53:49.000000000 +0300
+++ b/rc	2025-05-09 14:16:45.600715045 +0300
@@ -120,7 +120,7 @@ printinfo "\nDeleting temporary files.."
 
 # Set kernel variables
 printinfo "\nSetting kernel variables.."
-/sbin/sysctl -p > /dev/null || printf "%s" "${BOLD}${RED}[ERROR]${RESET}"
+/sbin/sysctl -e --system > /dev/null || printf "%s" "${BOLD}${RED}[ERROR]${RESET}"
 
 # Update shared library links
 printinfo "\nUpdating shared library links.."
