diff -Naur rustc-1.87.0-src.orig/compiler/rustc_target/src/spec/base/wasm.rs rustc-1.87.0-src/compiler/rustc_target/src/spec/base/wasm.rs
--- rustc-1.87.0-src.orig/compiler/rustc_target/src/spec/base/wasm.rs	2025-05-15 20:36:30.228688191 +0200
+++ rustc-1.87.0-src/compiler/rustc_target/src/spec/base/wasm.rs	2025-05-15 20:51:44.097660090 +0200
@@ -87,7 +87,7 @@
         limit_rdylib_exports: false,
 
         // we use the LLD shipped with the Rust toolchain by default
-        linker: Some("rust-lld".into()),
+        linker: Some("wasm-ld".into()),
         linker_flavor: LinkerFlavor::WasmLld(Cc::No),
 
         pre_link_args,
