--- a/src/prtget.cpp	2022-02-27 23:30:23.000000000 +0200
+++ b/src/prtget.cpp	2023-12-02 03:25:50.751321120 +0200
@@ -23,6 +23,7 @@ using namespace std;
 
 #include <sys/types.h>
 #include <sys/stat.h>
+#include <sys/utsname.h>
 #include <dirent.h>
 #include <unistd.h>
 
@@ -1640,6 +1641,17 @@ void PrtGet::fsearch()
 {
     assertMinArgCount(1);
 
+    // TODO: .footprint support multi arch riscv64 and aarch64
+    struct utsname sysinfo;
+    string suffix;
+
+    if ( uname(&sysinfo) == 0 ) {
+        string march = sysinfo.machine;
+        if ( march == "riscv64" || march == "aarch64" ) {
+            suffix = "." + march;
+        }
+    }
+
     string arg = "*";
     if ( m_parser->otherArgs().size() == 1 ) {
         arg = *(m_parser->otherArgs().begin());
@@ -1654,7 +1666,10 @@ void PrtGet::fsearch()
         string fp =
             it->second->path() + "/" +
             it->second->name() + "/" + ".footprint";
-        if ( File::grep( fp, arg, matches,
+        if ( File::grep( fp + suffix, arg, matches,
+                         m_parser->fullPath(),
+                         m_useRegex) ||
+             File::grep( fp, arg, matches,
                          m_parser->fullPath(),
                          m_useRegex)) {
             if ( matches.size() > 0 ) {
