--- a/Pkgfile
+++ b/Pkgfile
@@ -4,7 +4,7 @@
 # Depends on: gtk3 nss brotli alsa-lib xdg-utils cups libgcrypt ttf-liberation dbus pulseaudio pciutils libva libffi desktop-file-utils hicolor-icon-theme krb5 python3 gn ninja clang lld gperf nodejs git
 
 name=ungoogled-chromium
-version=124.0.6367.60
+version=124.0.6367.91
 _uc_version=$version-1
 release=1
 source=(https://commondatastorage.googleapis.com/chromium-browser-official/${name#*-}-$version.tar.xz
@@ -173,8 +173,11 @@ build() {
     _flags=(
       'custom_toolchain="//build/toolchain/linux/unbundle:default"'
       'host_toolchain="//build/toolchain/linux/unbundle:default"'
+      'clang_use_default_sample_profile=false'
+      'use_allocator="none"'
       'is_official_build=true' # implies is_cfi=true on x86_64
       'symbol_level=0' # sufficient for backtraces on x86(_64)
+      'is_cfi=false'
       'treat_warnings_as_errors=false'
       'disable_fieldtrial_testing_config=true'
       'blink_enable_generated_code_formatting=false'
@@ -191,9 +194,7 @@ build() {
       'enable_hangout_services_extension=true'
       'enable_widevine=true'
       'enable_nacl=false'
-      'enable_rust=true'
-      'rustc_version="yes"'
-      'rust_sysroot_absolute="/usr"'
+      'enable_rust=false'
       "google_api_key=\"$_google_api_key\""
     )
 
