--- a/dom/ipc/PContent.ipdl	2026-10-04 00:26:34.495775109 +0000
+++ b/dom/ipc/PContent.ipdl	2026-10-04 00:27:49.147619645 +0000
@@ -41,7 +41,7 @@ include protocol PURLClassifierLocal;
 include protocol PURLClassifierLocalByName;
 include protocol PVRManager;
 include protocol PRemoteMediaManager;
-#ifndef ANDROID
+#ifdef MOZ_WEBSPEECH
 include protocol PSpeechRecognition;
 #endif  // !ANDROID
 include protocol PRemoteWorkerDebuggerManager;
@@ -1200,7 +1200,7 @@ parent:
 
     async CreateAudioIPCConnection() returns (FileDescOrError fd);
 
-#ifndef ANDROID
+#ifdef MOZ_WEBSPEECH
     async AcquireHWInferenceProcess();
     async CreateSpeechRecognition(Endpoint<PSpeechRecognitionParent> endpoint);
 
--- a/ipc/glue/PUtilityProcess.ipdl	2026-10-04 01:45:29.074999659 +0000
+++ b/ipc/glue/PUtilityProcess.ipdl	2026-10-04 01:46:39.992824068 +0000
@@ -10,7 +10,7 @@ include protocol PProfiler;
 include protocol PUtilityMediaService;
 include protocol PJSOracle;
 
-#ifndef ANDROID
+#ifdef MOZ_WEBSPEECH
 include protocol PHWInference;
 #endif  // !ANDROID
 
