diff -Naur libreoffice-26.8.0.1.orig/Makefile.in libreoffice-26.8.0.1/Makefile.in
--- libreoffice-26.8.0.1.orig/Makefile.in	2026-07-16 13:53:49.461718502 +0200
+++ libreoffice-26.8.0.1/Makefile.in	2026-07-16 21:34:24.597174838 +0200
@@ -9,7 +9,7 @@
 
 gb_Top_MODULE_CHECK_TARGETS := slowcheck unitcheck subsequentcheck perfcheck uicheck screenshot
 
-.PHONY : check-if-root check-makefile-version bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo coverage internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
+.PHONY : check-makefile-version bootstrap gbuild build build-non-l10n-only build-l10n-only check clean clean-build clean-host test-install distclean distro-pack-install docs download etags fetch get-submodules id install install-gdb-printers install-strip tags debugrun help showmodules translations packageinfo coverage internal.clean $(gb_Top_MODULE_CHECK_TARGETS)
 
 MAKECMDGOALS?=build
 build_goal:=$(if $(filter build uicheck,$(MAKECMDGOALS)),build)\
@@ -67,18 +67,6 @@
 
 .DEFAULT_GOAL := build
 
-# Nested test is to check if the systemd-detect-virt command doesn't exist or it reports we're not
-# in a container:
-check-if-root:
-	@if test ! `uname` = 'Haiku' -a `id -u` = 0 && test -z $$container; then \
-		if test ! -x "$$(command -v systemd-detect-virt)" || ! systemd-detect-virt -c -q; then \
-			echo; \
-			echo 'Building LibreOffice as root is a very bad idea, use a regular user.'; \
-			echo; \
-			exit 1; \
-		fi \
-	fi
-
 check-makefile-version:
 	@if awk 'BEGIN {exit ($(MAKE_VERSION) >= 4.0) ;}' ; then \
 		echo; \
@@ -299,7 +287,7 @@
 #
 # Bootstrap
 #
-bootstrap: check-if-root check-makefile-version compilerplugins
+bootstrap: check-makefile-version compilerplugins
 
 #
 # Build
