--- a/Telegram/lib_base/base/platform/linux/base_url_scheme_linux.cpp	2025-09-23 17:28:03.260000000 +0000
+++ b/Telegram/lib_base/base/platform/linux/base_url_scheme_linux.cpp	2025-09-23 17:28:23.328000000 +0000
@@ -117,13 +117,13 @@ void RegisterUrlScheme(const UrlSchemeDe
 
 	const auto appId = QGuiApplication::desktopFileName().toStdString();
 	if (!appId.empty()) {
-		Gio::AppInfo appInfo = Gio::DesktopAppInfo::new_(appId + ".desktop");
+/*		Gio::AppInfo appInfo = Gio::DesktopAppInfo::new_(appId + ".desktop");
 		if (appInfo) {
 			if (appInfo.get_commandline() == commandlineForCreator + " %u") {
 				appInfo.set_as_default_for_type(handlerType);
 				return;
 			}
-		}
+		}*/
 	}
 
 	auto newAppInfo = Gio::AppInfo::create_from_commandline(
