--- a/cmake/external/minizip/CMakeLists.txt	2026-03-01 07:38:33.238129000 +0000
+++ b/cmake/external/minizip/CMakeLists.txt	2026-03-01 07:39:03.478129000 +0000
@@ -10,7 +10,7 @@ add_library(desktop-app::external_minizi
 if (DESKTOP_APP_USE_PACKAGED)
     find_package(PkgConfig)
     if (PkgConfig_FOUND)
-        pkg_check_modules(DESKTOP_APP_MINIZIP IMPORTED_TARGET minizip)
+        pkg_check_modules(DESKTOP_APP_MINIZIP IMPORTED_TARGET minizip<1.0.0)
     endif()
 
     if (DESKTOP_APP_MINIZIP_FOUND)
