diff -up mesa-25.1.0-rc1/src/intel/shaders/meson.build.3~ mesa-25.1.0-rc1/src/intel/shaders/meson.build
--- mesa-25.1.0-rc1/src/intel/shaders/meson.build.3~	2025-04-16 15:29:25.000000000 +0200
+++ mesa-25.1.0-rc1/src/intel/shaders/meson.build	2025-04-23 16:30:20.187354376 +0200
@@ -51,6 +51,8 @@ foreach gen : intel_shaders_gens
       '-I' + join_paths(dir_source_root, 'src/intel'),
       '-I' + join_paths(meson.project_build_root(), 'src/intel'),
       '-I' + join_paths(dir_source_root, 'src/intel/genxml'),
+      '-I/usr/lib64/clang/19/include',
+      '-I/usr/lib64/clang/20/include',
       cl_args,
     ],
     depends : [gen_cl_xml_pack, intel_dev_wa_src],
