diff -up mesa-25.2.0-rc1/src/gallium/drivers/panfrost/pan_screen.c.3~ mesa-25.2.0-rc1/src/gallium/drivers/panfrost/pan_screen.c
--- mesa-25.2.0-rc1/src/gallium/drivers/panfrost/pan_screen.c.3~	2025-07-16 16:42:48.000000000 +0200
+++ mesa-25.2.0-rc1/src/gallium/drivers/panfrost/pan_screen.c	2025-07-16 22:52:40.477061612 +0200
@@ -67,7 +67,7 @@ static const struct debug_named_value pa
    {"dirty",      PAN_DBG_DIRTY,      "Always re-emit all state"},
    {"sync",       PAN_DBG_SYNC,       "Wait for each job's completion and abort on GPU faults"},
    {"nofp16",     PAN_DBG_NOFP16,     "Disable 16-bit support"},
-   {"gl3",        PAN_DBG_GL3,        "Enable experimental GL 3.x implementation, up to 3.3"},
+   {"gl3",        PAN_DBG_NOGL3,      "Disable experimental GL 3.x implementation, up to 3.3"},
    {"noafbc",     PAN_DBG_NO_AFBC,    "Disable AFBC support"},
    {"nocrc",      PAN_DBG_NO_CRC,     "Disable transaction elimination"},
    {"linear",     PAN_DBG_LINEAR,     "Force linear textures"},
@@ -627,7 +627,7 @@ panfrost_init_screen_caps(struct panfros
    struct panfrost_device *dev = &screen->dev;
 
    /* Our GL 3.x implementation is WIP */
-   bool is_gl3 = dev->debug & PAN_DBG_GL3;
+   bool is_gl3 = (dev->debug & PAN_DBG_NOGL3) == 0;
 
    unsigned max_tib_size =
       pan_get_max_tib_size(dev->arch, dev->model);
diff -up mesa-25.2.0-rc1/src/panfrost/lib/pan_util.h.3~ mesa-25.2.0-rc1/src/panfrost/lib/pan_util.h
--- mesa-25.2.0-rc1/src/panfrost/lib/pan_util.h.3~	2025-07-16 16:42:48.000000000 +0200
+++ mesa-25.2.0-rc1/src/panfrost/lib/pan_util.h	2025-07-16 22:54:43.618132009 +0200
@@ -40,7 +40,7 @@
 /* 0x20 unused */
 #define PAN_DBG_NOFP16  0x0040
 #define PAN_DBG_NO_CRC  0x0080
-#define PAN_DBG_GL3     0x0100
+#define PAN_DBG_NOGL3   0x0100
 #define PAN_DBG_NO_AFBC 0x0200
 /* 0x400 unused */
 #define PAN_DBG_STRICT_IMPORT 0x0800
