diff -up mesa-26.0.0-rc3/meson.build.4~ mesa-26.0.0-rc3/meson.build
--- mesa-26.0.0-rc3/meson.build.4~	2026-02-06 18:53:13.995766819 +0100
+++ mesa-26.0.0-rc3/meson.build	2026-02-06 19:41:51.764842821 +0100
@@ -166,7 +166,7 @@ if gallium_drivers.contains('auto')
       gallium_drivers = [
         'v3d', 'vc4', 'freedreno', 'etnaviv', 'nouveau', 'svga',
         'tegra', 'virgl', 'lima', 'panfrost', 'llvmpipe', 'softpipe', 'iris',
-        'zink'
+        'zink', 'r300', 'r600', 'radeonsi'
       ]
     elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
       gallium_drivers = [
@@ -254,15 +254,13 @@ if _vulkan_drivers.contains('auto')
   # TODO: handle swrat driver if llvm is auto/disabled
   if system_has_kms_drm
     if host_machine.cpu_family().startswith('x86')
-      _vulkan_drivers = ['amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast']
+      _vulkan_drivers = ['amd', 'intel', 'intel_hasvk', 'nouveau', 'swrast', 'virtio']
     elif ['aarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['swrast', 'intel', 'panfrost', 'freedreno', 'asahi']
+      _vulkan_drivers = ['swrast', 'intel', 'panfrost', 'freedreno', 'asahi', 'amd', 'broadcom', 'virtio', 'imagination-experimental', 'nouveau']
     elif ['arm'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['swrast', 'intel', 'panfrost', 'freedreno']
-    elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
-    elif ['loongarch64'].contains(host_machine.cpu_family())
-      _vulkan_drivers = ['amd', 'swrast']
+      _vulkan_drivers = ['swrast', 'intel', 'panfrost', 'freedreno', 'amd', 'broadcom', 'virtio', 'imagination-experimental' , 'nouveau']
+    elif ['mips', 'mips64', 'ppc', 'ppc64', 'riscv32', 'riscv64', 'loongarch64'].contains(host_machine.cpu_family())
+      _vulkan_drivers = ['amd', 'swrast', 'intel', 'virtio', 'imagination-experimental', 'nouveau']
     else
       error('Unknown architecture @0@. Please pass -Dvulkan-drivers to set driver options. Patches gladly accepted to fix this.'.format(
             host_machine.cpu_family()))
