diff -up mesa-24.1.0-rc1/src/nouveau/compiler/meson.build.5~ mesa-24.1.0-rc1/src/nouveau/compiler/meson.build
--- mesa-24.1.0-rc1/src/nouveau/compiler/meson.build.5~	2024-04-24 22:28:09.000000000 +0200
+++ mesa-24.1.0-rc1/src/nouveau/compiler/meson.build	2024-04-26 04:26:38.283796090 +0200
@@ -25,11 +25,11 @@ nak_rust_args = [
   '-Anon_snake_case',
 ]
 
-dep_syn = dependency('syn',
-  version : '>= 2.0.15',
-  fallback : ['syn', 'dep_syn'],
-  required : true,
-)
+#dep_syn = dependency('syn',
+#  version : '>= 2.0.15',
+#  fallback : ['syn', 'dep_syn'],
+#  required : true,
+#)
 
 dep_paste = dependency('paste',
   version : '>= 1.0.14',
@@ -113,7 +113,13 @@ libnak_bindings_gen = static_library(
 _libnak_ir_proc_rs = rust.proc_macro(
   'nak_ir_proc',
   files('nak/ir_proc.rs'),
-  dependencies : [dep_syn],
+  #  dependencies : [dep_syn],
+  rust_args : [
+    '--extern', 'syn=../rustdeps/libsyn.rlib',
+    '--extern', 'proc_macro2=../rustdeps/libproc_macro2.rlib',
+    '--extern', 'quote=../rustdeps/libquote.rlib',
+    '-L', 'dependency=../rustdeps',
+  ]
 )
 
 _libnak_rs = static_library(
