From f9d8177f22baa1f218caafaf69b0af848d5cc03b Mon Sep 17 00:00:00 2001
From: Boris Brezillon <boris.brezillon@collabora.com>
Date: Wed, 15 Nov 2023 11:50:22 +0100
Subject: [PATCH 25/28] panfrost: Add a panfrost_model entry for G610

---
 src/panfrost/lib/pan_props.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/panfrost/lib/pan_props.c b/src/panfrost/lib/pan_props.c
index 6e4f58b6e4b..2bac6294a8c 100644
--- a/src/panfrost/lib/pan_props.c
+++ b/src/panfrost/lib/pan_props.c
@@ -71,6 +71,8 @@ const struct panfrost_model panfrost_model_list[] = {
         MODEL(0x7402, "G52 r1", "TGOx", HAS_ANISO,         16384, {}),
         MODEL(0x9091, "G57",    "TNAx", HAS_ANISO,         16384, {}),
         MODEL(0x9093, "G57",    "TNAx", HAS_ANISO,         16384, {}),
+
+        MODEL(0xa867, "G610",   "TNAx", HAS_ANISO,         16384, {}), // TODO
 };
 /* clang-format on */
 
-- 
2.43.0

