diff -up mesa-main/src/panfrost/lib/kmod/panthor_kmod.c.omv~ mesa-main/src/panfrost/lib/kmod/panthor_kmod.c
--- mesa-main/src/panfrost/lib/kmod/panthor_kmod.c.omv~	2024-01-14 18:41:28.934315439 +0100
+++ mesa-main/src/panfrost/lib/kmod/panthor_kmod.c	2024-01-14 18:41:51.354535016 +0100
@@ -373,7 +373,7 @@ panthor_kmod_bo_export(struct pan_kmod_b
    }
 
    bo->flags |= PAN_KMOD_BO_FLAG_EXPORTED;
-   return dmabuf_fd;
+   return 0;
 }
 
 static off_t
