--- a/config.guess	2023-04-22 08:16:15.896995151 +0000
+++ b/config.guess	2023-04-22 08:19:32.133681534 +0000
@@ -845,6 +845,12 @@ EOF
 	    echo ${UNAME_MACHINE}-unknown-linux-gnueabi
 	fi
 	exit ;;
+    aarch*:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
+    riscv*:Linux:*:*)
+	echo ${UNAME_MACHINE}-unknown-linux-gnu
+	exit ;;
     avr32*:Linux:*:*)
 	echo ${UNAME_MACHINE}-unknown-linux-gnu
 	exit ;;
